### PR TITLE
clear old context from contexts when secret is updated

### PR DESCRIPTION
**Pull request by ramaraochavali**
Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Commit Message: clear old context from contexts when secret is updated
Additional Description: When secret is updated via SDS, the old contexts still linger in the context list making the "/certs" render them and also "DaysUntilFirstCertExpiry" is incorrect - it reflects the replaced stats.
Risk Level: Low
Testing: Added
Docs Changes: N/A
Release Notes: N/A
Fixes https://github.com/envoyproxy/envoy/issues/8614

### CODE REVIEW

File: source/extensions/transport_sockets/tls/context_manager_impl.cc [hunk 1]
```diff
@@ -24,30 +24,44 @@ void ContextManagerImpl::removeEmptyContexts() {
   contexts_.remove_if([](const std::weak_ptr<Envoy::Ssl::Context>& n) { return n.expired(); });
 }
 
+void ContextManagerImpl::removeOldContext(std::shared_ptr<Envoy::Ssl::Context> old_context) {
+  if (old_context) {
+    contexts_.remove_if([old_context](const std::weak_ptr<Envoy::Ssl::Context>& n) {
+      std::shared_ptr<Envoy::Ssl::Context> sp = n.lock();
+      if (old_context && sp) {
```
Comment (rojkov) [inline source/extensions/transport_sockets/tls/context_manager_impl.cc:31]: Isn't `old_context` here redundant as it's checked in L28?

Comment (ramaraochavali) [reply source/extensions/transport_sockets/tls/context_manager_impl.cc:31]: Yeah. It is redundant

**Review by ggreenway — CHANGES_REQUESTED** [review]
This looks good overall. I think this makes the data from the admin intf be what people expect more.

Please update the comments in `include/envoy/ssl/context_manager.h` to clarify that `iterateContexts()` iterates through the contexts currently attached to a listener, not the currently allocated ones (which includes the ones only used by established connections).

**Comment by ramaraochavali** [comment]
@ggreenway Thank you. Updated the comment.

**Review by ggreenway — APPROVED** [review]

**Pull request merged**